Libraries/LibGUI/GTextEditor.cpp

```diff
@@ -1512,12 +1543,15 @@ void GTextEditor::RemoveCharacterCommand::undo()
 
 void GTextEditor::RemoveCharacterCommand::redo()
 {
-    //TOOD: Redo implementation
+    //Move back the cursor if it's inside in deleted content
+    if (m_text_editor.cursor().column() >= m_text_position.column())
+        m_text_editor.set_cursor(m_text_position.line(), m_text_position.column());
```

Comment:
Let's add a helper function on `GTextEditor` for moving the cursor to a guaranteed-safe position.
Some name ideas: `clamp_cursor_to_document()`, `ensure_cursor_is_valid()`, `move_cursor_to_valid_position_if_needed()`.